[DM-2498] "Replace device" wizard section (#1508)
SoftwareAG/c8y-docs: content/device-management-application/viewing-all-devices-bundle/to-replace-a-device.md

---
weight: 20
title: To replace a device
layout: redirect
---

> The device list offers a functionality to support the physical exchange of one device with another by preserving the original device representation on the platform including all its generated data. A wizard will guide you through the steps in this process. You can start the wizard by hovering over the device you want to replace in the device list and clicking the "Replace device" button.

Reviewer: ```suggestion
The device list supports the replacement of a physical device with another by preserving the original device representation in the platform including all its generated data. A wizard guides you through this process. Access the wizard by hovering over the device you want to replace in the device list and click the **Replace device** button.
```

> In the "Replacement device" step you will see the list of devices available on the platform excluding the device which will be replaced. You need to select the device which will physically replace the device you started the replacement process for.

Reviewer: ```suggestion
In the **Replacement device** step you can see the list of devices available in the platform excluding the device which you are going to replace. Select a replacement device from the list.
```

{{< c8y-admon-info >}}

> In order to find the replacement device in the list you need to first register it through the corresponding [registration process](/device-management-application/registering-devices/) appropriate for this device.

Reviewer: ```suggestion
In order for a particular replacement device to show up in the list, you must register it through the corresponding [registration process](/device-management-application/registering-devices/) for this device.
```

{{< /c8y-admon-info >}}

> The "Select external IDs" step will list the external IDs assigned to the replacement device. You can select the ones you want to be assigned to device representation of the replaced device. You must select at least one external ID. This will link the data your new physical devices sends to the original device representation object.

Reviewer: ```suggestion
The **Select external IDs** step lists the external IDs assigned to the replacement device. Select those you want assigned to the device representation of the replaced device. You must select at least one external ID. This links the data which your new physical devices sends to the original device representation object.
```

> Before proceeding with the actual replacement process you would need to confirm this action. Hitting "Replace" in the "Replace device" prompt will trigger the replacement.

Reviewer: ```suggestion
Before proceeding with the actual replacement, you must confirm this action. Click **Replace** in the **Replace device** dialog.
```

{{< c8y-admon-important >}}

> The replacement process will remove the selected replacement device from the {{< product-c8y-iot >}} database including all its generated data.

Reviewer: ```suggestion
The replacement process removes the original device from the {{< product-c8y-iot >}} database including all its generated data.
```

Author: The original device representation object will remain, the representation object of the replacement device will be deleted. Imagine that you physically replace Device A with Device B. On the platform "Device A" will be re-linked from Device A to Device B while all generated data is kept. "Device B" will be deleted after Device B is linked to "Device A" during the replacement. The data sent before being re-linked will be lost.

> The replacement process will remove the selected replacement device from the {{< product-c8y-iot >}} database including all its generated data.

Reviewer: above, it says that the generated data are preserved (line 7). this might be confusing - can you clarify a bit?

Author: https://github.com/SoftwareAG/c8y-docs/pull/1508/files#r1375704409

{{< /c8y-admon-important >}}

> The "Replace" step will display the progress of the replacement process and the result of every step executed.

Reviewer: ```suggestion
The **Replace** step displays the progress of the replacement process and the result of every step executed.
```

![Replace device](/images/users-guide/DeviceManagement/devmgmt-replace-device-wizard.png)<br>

> If a step has been skipped you will find an icon with a tooltip containg information about the reason for the step being skipped.

Reviewer: ```suggestion
If a step has been skipped you will find an icon with a tooltip which contains the reason why the step has been skipped.
```

> Should the execution of a given step have failed you will be able to see the error details by expanding step details. Note that an error in one of the steps will not prevent the next ones from an attempt to get executed. The wizard will try to execute as many steps as possible automatically and give you information about the failed ones. If the reason for the failure(s) can be removed you can retry all failed steps by clicking the "Retry" button in the bottom bar. If you wish to retry a single failed step only you can do so by clicking the "Retry this step" button form the step actions dropdown.

Reviewer: ```suggestion
If the execution of a given step has failed, you can see the error details by expanding the step details. Note that an error in one of the steps will not prevent the next steps from being executed. The wizard will try to execute as many steps as possible and provide information about failed steps. If you are able to resolve the reason for the failure(s), you can retry all failed steps by clicking the **Retry** button in the bottom bar. If you wish to retry a single failed step you can only do so by clicking the **Retry this step** button form the step actions dropdown.
```

> The "Close" button will bring you back the the updated device list. The replacement device will have been removed from {{< product-c8y-iot >}} while all data sent by the new physical device will be linked to the platform representation of the original physical device thus ensuring that the physical replacement would remain transparent to users.

Reviewer: ```suggestion
The **Close** button brings you back the the updated device list. The replacement device will have been removed from {{< product-c8y-iot >}} while all data sent by the new physical device will be linked to the platform representation of the original device. This ensures that the physical replacement remains transparent to users.
```

---
weight: 20
title: To replace a device
layout: redirect
---

The device list supports the replacement of a physical device with another by preserving the original device representation in the platform including all its generated data. A wizard guides you through this process. Access the wizard by hovering over the device you want to replace in the device list and click the **Replace device** button.

{{< c8y-admon-info >}}

> Currently LWM2M devices are not supported and the "Replace device" button is not available for this type of devices.

Reviewer: ```suggestion
Currently LWM2M devices are not supported and the **Replace device** button is not available for this type of devices.
```
assuming this is a UI label
